packages/rspack/tests/Stats.test.ts

```diff
@@ -177,4 +177,64 @@ describe("Stats", () => {
 		"
 	`);
 	});
+
+	it("should have reasons", async () => {
+		const stats = await compile({
+			context: __dirname,
+			entry: "./fixtures/main5.js",
+			stats: { reasons: true }
+		});
+		expect(stats?.toJson({}).modules).toMatchInlineSnapshot(`
```

Comment:
```suggestion
		expect(stats?.toJson({ all: false, reasons: true }).modules).toMatchInlineSnapshot(`
```

add `{ all: false, reasons: true }` can make the snapshot less noisy.